db.go

```diff
@@ -367,13 +371,6 @@ func (a dbAppender) Commit() error {
 // Old blocks are only deleted on reload based on the new block's parent information.
 // See DB.reload documentation for further information.
 func (db *DB) compact() (err error) {
-	db.cmtx.Lock()
```

Comment:
`db.cmtx` is also used in Delete, we should not remove it from here. Comments say `cmtx is used to control compactions and deletions.`.

Reply:
yes I see it now, these shouldn't happen at the same time.